linked-list.js

```diff
@@ -14,62 +14,188 @@ class LinkedList {
     this.head = null;
     this.tail = null;
     this.length = 0;
-
+    this.vals = vals
     for (let val of vals) this.push(val);
+
   }
 
   /** push(val): add new value to end of list. */
 
   push(val) {
-
+    let newNode = new Node(val)
+    if (this.head === null) {
+      this.head = newNode
+      this.tail = newNode
+    } else {
+      this.tail.next = newNode
+      this.tail = newNode
+    }
+    this.length++
```

Comment:
length is a kinda a weird property on a linkedlist since its not like you can directly go to an arbitrary node, not sure if this info in important